From fcf027c38ac94ded8cdb248ec41e9d37b32e8b14 Mon Sep 17 00:00:00 2001
From: Damjan Jovanovic <damjan@apache.org>
Date: Wed, 24 Oct 2012 19:01:22 +0000
Subject: [PATCH] Tiled TIFF images do not correctly load partial tiles.

Jira issue key: IMAGING-93
Submitted by: Gary Lucas <gwlucas at sonalysts dot com>



git-svn-id: https://svn.apache.org/repos/asf/commons/proper/imaging/trunk@1401825 13f79535-47bb-0310-9956-ffa450edef68
---
 src/changes/changes.xml                                       | 3 +++
 .../imaging/formats/tiff/datareaders/DataReaderTiled.java     | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index fc9117d2c..d836f1321 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -239,6 +239,9 @@ The <action> type attribute can be add,update,fix,remove.
       <action issue="IMAGING-95" dev="damjan" type="update" due-to="amitgupt">
          Some tiff processing takes very long.
       </action>
+      <action issue="IMAGING-93" dev="damjan" type="fix" due-to="gwlucas">
+         Tiled TIFF images do not correctly load partial tiles.
+      </action>
     </release>
     
   </body>
diff --git a/src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java b/src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java
index 592f6ae9c..0cc12438c 100644
--- a/src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java
+++ b/src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderTiled.java
@@ -83,13 +83,13 @@ private void interpretTile(ImageBuilder imageBuilder, byte bytes[],
             int i1 = startY + tileLength;
             if (i1 > height) {
                 // the tile is padded past bottom of image
-                i1 = height - startY;
+                i1 = height;
             }
             int j0 = startX;
             int j1 = startX + tileWidth;
             if (j1 > width) {
                 // the tile is padded to beyond the tile width
-                j1 = width - startX;
+                j1 = width;
             }
             if (photometricInterpreter instanceof PhotometricInterpreterRgb) {
                 for (int i = i0; i < i1; i++) {
